src/components/PeopleTable.tsx

```diff
@@ -1,4 +1,78 @@
-export const PeopleTable = () => {
+import { FC } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+import { SortBy } from '../types/SortBy';
+import { SortLink } from './SortLink';
+
+type Props = {
+  people: Person[],
+};
+
+export const PeopleTable: FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+  const [searchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+  const query = searchParams.get('query')?.toLowerCase() || '';
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const filteredPeople = people.filter(person => {
+    if (sex && person.sex !== sex) {
+      return false;
+    }
+
+    if (centuries.length) {
+      const century = Math.ceil(person.born / 100);
+
+      if (!centuries.includes(`${century}`)) {
+        return false;
+      }
+    }
+
+    if (query
+      && !(
+        person.name.toLowerCase().includes(query)
+        || person.motherName?.toLowerCase().includes(query)
+        || person.fatherName?.toLowerCase().includes(query)
+      )
+    ) {
+      return false;
+    }
+
+    return true;
+  });
+
+  const onSort = () => {
+    switch (sort) {
```

Comment:
Instead of using an arrow function for 'onSort', consider using a regular function to avoid problems with 'this' binding